core/src/main/scala/cr/pulsar/schema/Schema.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Chatroulette
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cr.pulsar.schema
+
+import cr.pulsar.Consumer.DecodingFailure
+
+import cats.Inject
+import org.apache.pulsar.client.api.{ Schema => JSchema }
+import org.apache.pulsar.common.schema.SchemaInfo
+
+trait Schema[A] {
+  def get: JSchema[A]
```

Comment:
I would call it in a different way, eg `schema`. Otherwise in the code it looks confusing:
```client.newProducer(E.get).topic(topic.url.value)```

When I saw this I was surprised that we are getting something. Also, usually in Scala, it's a flag that something shady is going on (eg `Option.get`)

Reply:
Makes sense 👍🏾